src/cc/common.cc

```diff
@@ -47,5 +49,43 @@ std::vector<int> get_possible_cpus() {
   return read_cpu_range("/sys/devices/system/cpu/possible");
 }
 
+FileDesc::FileDesc(int fd) : fd_(fd) {}
+
+FileDesc::FileDesc(FileDesc &&that) {
+  *this = std::move(that);
+}
+
+FileDesc::~FileDesc() {
+  if (fd_ >= 0)
+    ::close(fd_);
+}
+
+FileDesc &FileDesc::operator=(int fd) {
+  if (fd_ >= 0)
+    ::close(fd_);
+  fd_ = fd;
+  return *this;
+}
+
+FileDesc &FileDesc::operator=(FileDesc &&that) {
+  if (fd_ >= 0)
```

Comment:
Added cleanup in move constructor and `int` assign operator since this class should be more generic now.